From d058988e1f2a67769075066e875cd8b16de931a4 Mon Sep 17 00:00:00 2001
From: Quentin Armitage <quentin@armitage.org.uk>
Date: Thu, 20 Aug 2020 17:06:19 +0100
Subject: [PATCH 322/691] Fix detecting setsid() error in xdaemon()

setsid() return should be checked for < 0 rather than < -1. This
error was fixed in the zebra code in 2004!

Signed-off-by: Quentin Armitage <quentin@armitage.org.uk>
---
 keepalived/core/daemon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/keepalived/core/daemon.c b/keepalived/core/daemon.c
index 0e3621d..3f99d34 100644
--- a/keepalived/core/daemon.c
+++ b/keepalived/core/daemon.c
@@ -59,7 +59,7 @@ xdaemon(bool nochdir, bool noclose, bool exitflag)
 
 	/* Become session leader and get pid. */
 	pid = setsid();
-	if (pid < -1) {
+	if (pid < 0) {
 		log_message(LOG_INFO, "xdaemon: setsid error");
 		return -1;
 	}
-- 
1.8.3.1

